src/mpyl/steps/run_properties.py

```diff
@@ -0,0 +1,77 @@
+"""Module to initiate run properties"""
+import logging
+from pathlib import Path
+from typing import Optional
+
+from ..cli import MpylCliParameters
+from ..project import load_project, Stage, Project
+from ..stages.discovery import find_build_set
+from ..steps.models import RunProperties
+from ..utilities.repo import Repository, RepoConfig
+
+
+def initiate_run_properties(
+    config: dict,
+    properties: dict,
+    cli_parameters: MpylCliParameters = MpylCliParameters(),
+    run_plan: Optional[dict[Stage, set[Project]]] = None,
+    all_projects: Optional[set[Project]] = None,
+) -> RunProperties:
+    if all_projects is None or run_plan is None:
+        with Repository(RepoConfig.from_config(config)) as repo:
+            if all_projects is None:
+                project_paths = repo.find_projects()
+                all_projects = set(
+                    map(
+                        lambda p: load_project(
+                            root_dir=Path(""),
```

Comment:
The root dir needs to be passed in from the outside, otherwise it's a pointless parameter.